File: backend/src/main/java/com/ams/entity/Manufacturer.java
package com.ams.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("manufacturer")
public class Manufacturer {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String code;
    private String contact;
    private String phone;
    private String email;
    private String website;
    private String country;
    private String address;
    private Integer status;
    private String remark;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/manufacturer.ts
import http from '@/utils/http';

export interface Manufacturer {
  id?: number;
  name: string;
  code?: string;
  contact?: string;
  phone?: string;
  email?: string;
  website?: string;
  country?: string;
  address?: string;
  status?: number;
  remark?: string;
}

export const getManufacturers = (params?: Record<string, any>) =>
  http.get<any>('/manufacturers', { params });

export const getManufacturerOptions = () =>
  http.get<Manufacturer[]>('/manufacturers/options');

export const getManufacturerDetail = (id: number) =>
  http.get<Manufacturer>(`/manufacturers/${id}`);

export const createManufacturer = (data: Manufacturer) =>
  http.post<Manufacturer>('/manufacturers', data);

export const updateManufacturer = (id: number, data: Manufacturer) =>
  http.put<Manufacturer>(`/manufacturers/${id}`, data);

export const deleteManufacturer = (id: number) =>
  http.delete(`/manufacturers/${id}`);
